crates/cairo-addons/src/vm/hint_definitions/maths.rs

```diff
@@ -0,0 +1,120 @@
+    Felt252,
+};
+
+use crate::vm::hints::Hint;
+
+pub const HINTS: &[fn() -> Hint] = &[felt252_to_bytes_le, felt252_to_bytes_be];
+
+pub fn felt252_to_bytes_le() -> Hint {
+    Hint::new(
+        String::from("felt252_to_bytes_le"),
+        |vm: &mut VirtualMachine,
+         _exec_scopes: &mut ExecutionScopes,
+         ids_data: &HashMap<String, HintReference>,
+         ap_tracking: &ApTracking,
+         _constants: &HashMap<String, Felt252>|
+         -> Result<(), HintError> {
+            // Get input values from Cairo
+            let value = get_integer_from_var_name("value", vm, ids_data, ap_tracking)?;
+            let len = get_integer_from_var_name("len", vm, ids_data, ap_tracking)?;
+            let output_ptr = get_ptr_from_var_name("output", vm, ids_data, ap_tracking)?;
+
+            let len: usize =
+                len.try_into().map_err(|_| MathError::Felt252ToUsizeConversion(Box::new(len)))?;
+
+            let truncated_value = if len < 32 {
+                // Create mask for truncation: (1 << (len * 8)) - 1
+                let mask = (1_u128 << (len * 8)) - 1;
+                felt252_bit_and(value, mask.into())?
+            } else {
+                value
+            };
+
+            // Convert to bytes and write to memory
+            let bytes = truncated_value
+                .to_bytes_le()
+                .split_at(len)
+                .0
+                .iter()
+                .map(|b| MaybeRelocatable::Int((*b).into()))
+                .collect::<Vec<MaybeRelocatable>>();
+            vm.segments.write_arg(output_ptr, &bytes)?;
+            Ok(())
+        },
+    )
+}
+
+pub fn felt252_to_bytes_be() -> Hint {
+    Hint::new(
+        String::from("felt252_to_bytes_be"),
+        |vm: &mut VirtualMachine,
+         _exec_scopes: &mut ExecutionScopes,
+         ids_data: &HashMap<String, HintReference>,
+         ap_tracking: &ApTracking,
+         _constants: &HashMap<String, Felt252>|
+         -> Result<(), HintError> {
+            // Get input values from Cairo
+            let value = get_integer_from_var_name("value", vm, ids_data, ap_tracking)?;
+            let len = get_integer_from_var_name("len", vm, ids_data, ap_tracking)?;
+            let output_ptr = get_ptr_from_var_name("output", vm, ids_data, ap_tracking)?;
+
+            let len: usize =
+                len.try_into().map_err(|_| MathError::Felt252ToUsizeConversion(Box::new(len)))?;
+
+            let truncated_value = if len < 32 {
+                // Create mask for truncation: (1 << (len * 8)) - 1
+                let mask = (1_u128 << (len * 8)) - 1;
+                felt252_bit_and(value, mask.into())?
+            } else {
+                value
+            };
+
+            // Convert to bytes and write to memory
+            let bytes = truncated_value
+                .to_bytes_be()
+                .split_at(32 - len)
+                .1
+                .iter()
```

Comment:
can probably use into_iter() to own the collection to avoid copying it, though not sure itd work